src/libra_py/pdos.py

```diff
@@ -384,4 +384,154 @@ def libra_pdos(_emin, _emax, _de, projections, prefix, outfile, Nel, do_convolve
+
+    
+    # Finding the homo and lumo energy level by appending the 
+    # pdos numerical values of unoccupied states only
+    pdos_unocc = []
+    # Energy levels
+    energy_levels = []
+    for i in range(first_index, last_index + 1):
+        energy_levels.append(float(lines[i].split()[1])*energy_conversion)
+        if float(lines[i].split()[2])==0:
+            pdos_unocc.append(i)
+    # LUMO energy level
+    lumo_level = int(lines[min(pdos_unocc)].split()[0])
+    # HOMO energy level
+    homo_level = lumo_level-1
+    # HOMO energy
+    homo_energy = float(lines[homo_level].split()[1])*energy_conversion
+    # Minimum energy level
+    min_energy = float(lines[first_index].split()[1])*energy_conversion
+    # Maximum energy level
+    max_energy = float(lines[last_index].split()[1])*energy_conversion
+    
+    
+    # Now we make an equispaced energy vector from min_energy ad max_energy with npoints.
+    energy_grid = np.linspace( min_energy-2, max_energy+2, npoints )
+    energy_grid = np.array(energy_grid)
+    
+    
+    # Appending the energy lines with their component densities of states
+    energy_lines = []
+    for i in range( time_step * ( num_levels + 2 ) + 2, ( time_step + 1 ) * ( num_levels + 2 ) ):
+        # Appending the energy lines into enrgy_lines
+        energy_lines.append( lines[i].split() )
+
+    for i in range(0, len(energy_lines)):
+        
+        for j in range(0,len(energy_lines[0])):
+            
+            energy_lines[i][j] = float(energy_lines[i][j])
+            
+    energy_lines = np.array(energy_lines)
+
+    # Now we sum the PDOSs defined in angular_momentum_cols by user
+    pdos_sum = []
+    for k in range(0, len(energy_lines)):
+        
+        # A temporary vector for summation of the PDOS
+        tmp_vec = []
+        tmp_vec.append(energy_lines[k][1])
+        
+        for i in range(0,len(angular_momentum_cols)):
+            # Initializing a new sum variable
+            # print("angular_momentum_cols[i]",angular_momentum_cols[i])
+            tmp_sum = 0
+            for j in angular_momentum_cols[i]:
+                
+                # If j is less than the number of columns 
+                # then sum the PDOS
+                if j<=num_cols:
+                    tmp_sum += energy_lines[k][j]
+            # Appending tmp_sum into tmp_vec
+            tmp_vec.append(tmp_sum)
+        
+        # Now append tmp_vec into pdos_sum, we will
+        # then use this pdos_sum for convolution
+        pdos_sum.append(tmp_vec)
+    
+    convolved_pdos = []
+    t1 = time.time()
+    for j in range(1,len(angular_momentum_cols)+1):
+        # Initialize a vector of zeros summing the weighted PDOS
+        tmp_weighted_pdos = np.zeros(energy_grid.shape)
+
+        for i in range(0,num_levels):
+            # The Guassian function
+            gaussian_fun = (coef/(sigma*np.sqrt(2.0*np.pi)))*(np.exp(-0.5*np.power(((energy_grid-float(pdos_sum[i][0])*energy_conversion)/sigma),2)))

```

Comment:
perhaps, pre-compute the common pre-factor before the double loop to reduce the computational time